nav2_system_tests/src/planning/planner_tester.cpp

```diff
@@ -59,22 +69,102 @@ PlannerTester::PlannerTester()
   startCostmapServer();
 
   // Launch a thread to process the messages for this node
-  spin_thread_ = new std::thread(&PlannerTester::spinThread, this);
+  spin_thread_ = std::make_unique<std::thread>(
+    [&]()
+    {
+      executor_.add_node(this->get_node_base_interface());
+      executor_.spin();
+      executor_.remove_node(this->get_node_base_interface());
+    });
 }
 
-PlannerTester::~PlannerTester()
+void PlannerTester::deactivate()
 {
+  if (!is_active_) {
+    RCLCPP_WARN(this->get_logger(), "Trying to deactivate while already inactive");
+    return;
+  }
+  is_active_ = false;
+
   executor_.cancel();
   spin_thread_->join();
-  delete spin_thread_;
-  spin_thread_ = nullptr;
+  spin_thread_.reset();
+
+  planner_client_.reset();
+  map_timer_.reset();
+  map_pub_.reset();
+  map_.reset();
+  costmap_server_.reset();
+  transform_publisher_.reset();
+}
+
+PlannerTester::~PlannerTester()
+{
+  if (is_active_) {
+    deactivate();
+  }
+}
+
+void PlannerTester::startPoseTransform()
+{
+  // Provide the robot pose transform
+  transform_publisher_ = create_publisher<tf2_msgs::msg::TFMessage>("/tf", rclcpp::QoS(100));
```

Comment:
This should probably be a tf2_ros::TransformBroadcaster rather than a generic publisher.

Reply:
Agreed, though a Broadcaster is just a `/tf` publisher https://github.com/ros2/geometry2/blob/ros2/tf2_ros/include/tf2_ros/transform_broadcaster.h#L58

Reply:
Switched to a Broadcaster.